docs/notebook-demos/map-algebra.ipynb

```diff
@@ -0,0 +1,655 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "import geopyspark as gps\n",
+    "import numpy as np\n",
+    "\n",
+    "from pyspark import SparkContext\n",
+    "from shapely.geometry import Point, MultiPolygon, LineString, box"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "conf = gps.geopyspark_conf(master=\"local[*]\", appName=\"map-algebra\")\n",
+    "pysc = SparkContext(conf=conf)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# Setting up the data\n",
+    "\n",
+    "cells = np.array([[[3, 4, 1, 1, 1],\n",
+    "                   [7, 4, 0, 1, 0],\n",
+    "                   [3, 3, 7, 7, 1],\n",
+    "                   [0, 7, 2, 0, 0],\n",
+    "                   [6, 6, 6, 5, 5]]], dtype='int32')\n",
+    "\n",
+    "extent = gps.ProjectedExtent(extent = gps.Extent(0, 0, 5, 5), epsg=4326)\n",
+    "\n",
+    "layer = [(extent, gps.Tile.from_numpy_array(numpy_array=cells))]\n",
+    "  \n",
+    "rdd = pysc.parallelize(layer)\n",
+    "raster_layer = gps.RasterLayer.from_numpy_rdd(gps.LayerType.SPATIAL, rdd)\n",
+    "tiled_layer = raster_layer.tile_to_layout(layout=gps.LocalLayout(tile_size=5))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Map Algebra\n",
+    "\n",
+    "Given a set of raster layers, it may be desirable to combine and filter the content of those layers.  This is the function of _map algebra_.  Two classes of map algebra operations are provided by GeoPySpark: _local_ and _focal_ operations.  Local operations individually consider the pixels or cells of one or more rasters, applying a function to the corresponding cell values.  For example, adding two rasters' pixel values to form a new layer is a local operation.\n",
```

Comment:
I'm not sure if an introduction to map algebra is needed because of GeoPySpark's target audience should be somewhat familiar with the concept.